public/src/client/chats.js

```diff
@@ -530,6 +532,72 @@ define('forum/chats', ['components', 'string', 'sounds', 'forum/infinitescroll',
 			callback();
 		});
 	}
+	
+	Chats.handleSearch = function() {
+		var timeoutId = 0;
+		
+		components.get('chat/search').on('keyup', function() {
+			if (timeoutId) {
+				clearTimeout(timeoutId);
+				timeoutId = 0;
+			}
 
+			timeoutId = setTimeout(doSearch, 250);	
+		});
+		
+		function doSearch() {
+            		var username = components.get('chat/search').val();
+            		var chatsListEl = $('[component="chat/search/list"]');
+            
+            		if(username) {
```

Comment:
This is just a style change but I prefer to have the check here instead of 50 lines down.

So it would be

```
if (!username) {
    return chatsListEl.empty();
}
//rest of code
```